internal/product/product.go

```diff
@@ -60,8 +68,10 @@ func (s *Service) GetProduct(ID uint) (Product, error) {
 // PostProduct - adds a new product to the database
 func (s *Service) PostProduct(product Product) (Product, error) {
 	if result := s.DB.Save(&product); result.Error != nil {
-		log.Error("PostProduct error")
-		return Product{}, result.Error
+		log.WithFields(log.Fields{
+			"Function": "PostProduct",
+		}).Error("Product can't sending")
```

Comment:
Error(result.Error.Error()) return like this, this way you can store the exact error cause